kingpin/actors/rightscale/api.py

```diff
@@ -513,18 +532,28 @@ def run_executable_on_instances(self, name, inputs, instances):
 
         # Generate all the tasks and store them in a list so that we can yield
         # them all at once (thus, making it asynchronous)
+        task_pairs = []
         tasks = []
         for i in instances:
             log.debug('Executing %s on %s' % (name, i.soul['name']))
             url = '%s/run_executable' % i.links['self']
-            tasks.append(self.make_generic_request(url, post=params))
+            req = self.make_generic_request(url, post=params)
+            task_pairs.append((i, req))
+            tasks.append(req)
 
         try:
-            ret = yield tasks
+            yield tasks
         except requests.exceptions.HTTPError as e:
             raise ServerArrayException('Script Execution Error: %s' % e)
 
-        raise gen.Return(ret)
+        # At this point all the tasks are executed, but the reference
+        # that we store in task_pairs are still to the Future objects.
+        yielded_tasks = []
+        for (i, task) in task_pairs:
+            result = yield task  # Should be a no-op since it's completed above
+            yielded_tasks.append((i, result))
+
+        raise gen.Return(yielded_tasks)
```

Comment:
You've changed the return value here, so the docstrings above are wrong. Please update them.